src/evo/deterministicmns.cpp

```diff
@@ -1292,6 +1292,32 @@ static std::optional<ProTx> GetValidatedPayload(const CTransaction& tx, gsl::not
     return opt_ptx;
 }
 
+bool IsVersionChangeValid(gsl::not_null<const CBlockIndex*> pindexPrev, const uint16_t tx_type,
```

Comment:
```suggestion
  /**
   * Validates whether a masternode ProTx version change is permitted after V23 deployment.
   * @param pindexPrev Previous block index for deployment checking
   * @param tx_type Transaction type (TRANSACTION_PROVIDER_UPDATE_SERVICE, etc.)
   * @param state_version Current masternode state version
   * @param tx_version Proposed transaction version
   * @param state Validation state for error reporting
   * @return true if version change is valid, false otherwise
   */
bool IsVersionChangeValid(gsl::not_null<const CBlockIndex*> pindexPrev, const uint16_t tx_type,
```